lib/nanoc/base/entities/configuration.rb

```diff
@@ -47,6 +47,18 @@ def with_defaults
       self.class.new(new_wrapped)
     end
 
+    def with_environment(env)
```

Comment:
Instead of passing `env` as an argument (here and in `new_from_cwd`) would it be correct instead to:
- have in Nanoc::CLI
```
def self.env=(environment)
   ENV.store 'NANOC_ENVIRONMENT', environment
end
```
- have `with_environment` to fetch `NANOC_ENVIRONMENT` in ENV